### PR TITLE
show code lens

### PR DESCRIPTION
**Pull request by jm-observer**
Show code lens, but the click event has not been implemented

the effect as follow:

https://github.com/user-attachments/assets/9c6f7d98-1861-4715-b2bb-528df42392fe

### CODE REVIEW

**Comment by dzhou121** [comment]
I'm wondering if it makes sense to reuse the "code action" popup for the codelens popup?

**Comment by jm-observer** [comment]
I refer to the implementation of the code action. Compared with code lens, it does not use overlay, has the function of turning pages, and seems to have the function of shortcut keys, so the implementation is more complicated for me.

**Comment by jm-observer** [comment]
Now there's a problem with code lens. In addition to clicking to cancel the display, what scenarios should I cancel the display? How to achieve more appropriate?

**Comment by jm-observer** [comment]
In addition, the code action can only display one icon at a time, and the icon does not always exist. but code lens, on the other hand, shows more than one at a time and is always there. They are not similar

**Comment by dzhou121** [comment]
I think when you click another codelens icon, it should cancel the previous one if there was one. That's also why I suggested to reuse the code actions popup, so that it's guaranteed to always have one popup display at all time and the style is consistent as well.

The codelens icon is shown all the time but the popup for it doesn't.

**Comment by jm-observer** [comment]
Wait until i am more familiar with the code action

**Comment by jm-observer** [comment]
the effect as follow:
https://github.com/user-attachments/assets/6cddaca9-75d3-4781-bd92-69e535b5e77b

File: lapce-app/src/window_tab.rs [hunk 1]
```diff
@@ -1924,7 +1926,7 @@ impl WindowTabData {
                     // todo filter by language
                     self.main_split.docs.with_untracked(|x| {
                         for doc in x.values() {
-                            doc.get_semantic_styles();
+                            doc.on_update(None);
```
Comment (dzhou121) [inline lapce-app/src/window_tab.rs:1929]: not a good idea to call `on_update` here, and I would keep `on_update` private.

**Pull request merged**